### PR TITLE
feat: add freezing backbone arg in training script

### PR DESCRIPTION
**Pull request by charlesmindee**
This PR adds a freeze_bckb arg in the detection training script to freeze resnet50 when training db net.
Any feedback is welcome!

### CODE REVIEW

File: doctr/models/detection/differentiable_binarization.py [hunk 1]
```diff
@@ -524,7 +524,13 @@ def call(
         return dict(proba_map=proba_map)
```
Comment (fg-mindee) [inline doctr/models/detection/differentiable_binarization.py:524]: no need to make modifications to differentiable_binarization!

File: references/detection/train.py [hunk 2]
```diff
@@ -126,7 +126,12 @@ def main(args):
     optimizer = tf.keras.optimizers.Adam(learning_rate=args.lr, clipnorm=5)
 
     # Load doctr model
-    model = detection.__dict__[args.model](pretrained=False, input_shape=(args.input_size, args.input_size, 3))
+    model = detection.__dict__[args.model](
+        pretrained=False,
+        freeze_bckb=args.freeze_bckb,
+        input_shape=(args.input_size, args.input_size, 3)
+    )
+
```
Comment (fg-mindee) [inline references/detection/train.py:134]: you can roll that back, we can modify this afterwards

File: references/detection/train.py [hunk 3]
```diff
@@ -193,6 +198,7 @@ def parse_args():
     parser.add_argument('-j', '--workers', type=int, default=4, help='number of workers used for dataloading')
     parser.add_argument('--resume', type=str, default=None, help='Path to your checkpoint')
     parser.add_argument("--test-only", dest='test_only', action='store_true', help="Run the validation loop")
+    parser.add_argument('--freeze_bckb', type=bool, default=False, help='freeze model backbone for fine-tuning')
```
Comment (fg-mindee) [inline references/detection/train.py:201]:
```
parser.add_argument('--freeze-backbone', dest='freeze_backbone', action='store_true', help='freeze model backbone for fine-tuning')
```

File: references/detection/train.py (hunk 2 above)
Comment (fg-mindee) [reply references/detection/train.py:134]: Right before the training loop & loss queue, add this:
```
if args.freeze_backbone:
    for layer in model.feat_extractor.layers:
        layer.trainable = False
```

**Review by fg-mindee — CHANGES_REQUESTED** [review]
Thanks for the PR! I suggested a lighter change that would yield the same result without changing the DB implementation :ok_hand:

File: test/test_models_detection.py [hunk 4]
```diff
@@ -116,6 +116,7 @@ def test_detection_models(arch_name, input_shape, output_size, out_prob):
         assert np.all(np.logical_and(out >= 0, out <= 1))
 
 
+
```
Comment (fg-mindee) [inline test/test_models_detection.py]: flake is going to say there are too many blank lines :sweat_smile:

**Review by fg-mindee — APPROVED** [review]
Thanks!
